### PR TITLE
Rename BindNameInfo to EntityName

### PR DESCRIPTION
**Pull request by geoffromer**

### CODE REVIEW

**Comment by chandlerc** [comment]
FWIW, happy to review this and other rename PRs once there is consensus on a new name, but not sure we're there yet -- similar to the message on #4091

**Comment by geoffromer** [comment]
I think `EntityName` has consensus.

File: toolchain/sem_ir/bind_name.h [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 namespace Carbon::SemIR {
 
-struct BindNameInfo : public Printable<BindNameInfo> {
+struct EntityName : public Printable<EntityName> {
```
Comment (jonmeow) [inline toolchain/sem_ir/bind_name.h:15]: Should this file be renamed to match?

File: toolchain/sem_ir/ids.h [hunk 2]
```diff
@@ -174,12 +174,12 @@ constexpr ConstantId ConstantId::Error =
     ConstantId::ForTemplateConstant(InstId::BuiltinError);
 constexpr ConstantId ConstantId::Invalid = ConstantId(InvalidIndex);
 
-// The ID of a bind name.
-struct BindNameId : public IdBase, public Printable<BindNameId> {
-  using ValueType = BindNameInfo;
+// The ID of a EntityName.
+struct EntityNameId : public IdBase, public Printable<EntityNameId> {
```
Comment (jonmeow) [inline toolchain/sem_ir/ids.h:198]: Should the label in Print be updated?

File: toolchain/check/import_ref.cpp [hunk 3]
```diff
@@ -859,15 +859,15 @@ class ImportRefResolver {
     }
 
     const auto& import_bind_info =
```
Comment (jonmeow) [inline toolchain/check/import_ref.cpp]: Should this be renamed to match?

**Review by jonmeow — COMMENTED** [review]
Looks good, just highlighting a few references to BindNameInfo that I think might've been missed.

**Review by jonmeow — APPROVED** [review]

**Pull request merged**